orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy

```diff
@@ -266,7 +266,16 @@ class OperationsController {
 
   private void decorateBuildInfo(Map trigger) {
     if (trigger.master && trigger.job && trigger.buildNumber) {
-      def buildInfo = buildService.getBuild(trigger.buildNumber, trigger.master, trigger.job)
+      def buildInfo
+      try {
+        buildInfo = buildService.getBuild(trigger.buildNumber, trigger.master, trigger.job)
+      } catch (RetrofitError e) {
+        if (e.response?.status == 404) {
+          throw new IllegalStateException("Build " + trigger.buildNumber + " for " + trigger.master + "/" + trigger.job + " not found")
+        } else {
+          throw e
```

Comment:
i think it'd be nicer to do something like:
`throw new IllegalStageException("Failed to get the build ${...}...", e)`
that way we don't leak random exception without explanation.

(also this would be for `catch (Exception e)`

Reply:
Do you mean instead of just throwing the exception without changing the message, adding some explanation?

Reply:
yeah, say, you get retrofit error `Gateway Notfound`. If we just throw it as is, the UI will show `Gateway notfound` and the user will be scratching their head, wondering what does that mean. But, if we instead throw "We failed to get your build, btw, here is your inner exception:..." that gives them a ton more information. Also for anyone on support diagnosing will be pointed to the right place much faster with extra context

Reply:
Sounds good, I'll add that.